Assets/Scripts/Gameplay/Action/BaseTypes/Action.cs

```diff
@@ -29,23 +29,18 @@ namespace Unity.Multiplayer.Samples.BossRoom.Actions
     /// After shutdown: ChainIntoNewAction()    (only if Action was blocking, and only if End() was called, not Cancel())
     ///
     /// Note also that if Start() returns false, no other functions are called on the Action, not even End().
+    ///
+    /// This Action system has not been designed to be generic and extractable to be reused in other projects - keep that in mind when reading through this code.
```

Comment:
```suggestion
    /// This Action system has not been designed to be generic and extractable to be reused in other projects - keep that in mind when reading through this code.
    /// A better action system would need to be more accessible and customizable by game designers and allow more design emergence. It'd have ways to define smaller atomic action steps and have a generic way to define and access character data. It would also need to be more performant, as actions would scale with your number of characters and concurrent actions.
```